platform/core/src/types/HangingProtocol.ts

```diff
@@ -1,63 +1,86 @@
-type DisplaySetInfo = {
-  SeriesInstanceUID: string;
-  displaySetInstanceUID: string;
-  displaySetOptions: Record<string, unknown>;
+export type DisplaySetInfo = {
+  displaySetInstanceUID?: string;
+  displaySetOptions: DisplaySetOptions;
 };
 
-type ViewportMatchDetails = {
+export type ViewportMatchDetails = {
   viewportOptions: ViewportOptions;
   displaySetsInfo: DisplaySetInfo[];
 };
 
-type DisplaySetMatchDetails = {
-  SeriesInstanceUID: string;
-  StudyInstanceUID: string;
+export type DisplaySetMatchDetails = {
+  StudyInstanceUID?: string;
   displaySetInstanceUID: string;
   matchDetails?: any;
-  matchingScores?: any[];
+  matchingScores?: DisplaySetMatchDetails[];
   sortingInfo?: any;
 };
 
-
-type DisplaySetAndViewportOptions = {
+export type DisplaySetAndViewportOptions = {
   displaySetInstanceUIDs: string[];
   viewportOptions: ViewportOptions;
   displaySetOptions: DisplaySetOptions;
-}
-
-type ViewportSpecificProtocolOptions = {
-  [viewportIndex: string]: DisplaySetAndViewportOptions
-}
-
-type GlobalProtocolOptions = DisplaySetAndViewportOptions
+};
 
+export type SetProtocolOptions = {
+  // Contains a map of reuseId values to display set UIDs
+  reuseIdMap?: Record<string, string>;
 
-type SetProtocolOptions =
-  ViewportSpecificProtocolOptions | GlobalProtocolOptions;
+  // Indicates a stage to apply:
+  // The stage can be specified by id or by index
+  // TODO - make the stageId a regexp to allow searching for first matching
+  stageId?: string;
+  stageIndex?: number;
 
+  // Indicates to setup the protocol and fire the PROTOCOL_RESTORED event
+  // but don't fire the protocol changed event.  Used to restore the
+  // HP service to a previous state
+  restoreProtocol?: boolean;
```

Comment:
how about `restoreWithoutChangeEvent` since it is basically control over event

Reply:
That is what it is right now, but it could do things differently in the future - for example, having a cached set of changes, so I think restoreProtocol is actually descriptive.